File: app/src/main/java/com/ternaryop/photoshelf/adapter/OnPhotoBrowseClickMultiChoice.java
package com.ternaryop.photoshelf.adapter;

/**
 * Allow adapter to listen for clicks used by multi choice views
 */
public interface OnPhotoBrowseClickMultiChoice extends OnPhotoBrowseClick {
    /**
     * The click occurs on entire row
     * @param position the clicked item position
     */
    void onItemClick(int position);

    /**
     * The long click occurs on entire row
     * @param position the clicked item position
     */
    void onItemLongClick(int position);
}

File: core/src/main/java/com/ternaryop/photoshelf/adapter/OnPhotoBrowseClickMultiChoice.kt
package com.ternaryop.photoshelf.adapter

/**
 * Allow adapter to listen for clicks used by multi choice views
 */
interface OnPhotoBrowseClickMultiChoice : OnPhotoBrowseClick {
    /**
     * The click occurs on entire row
     * @param position the clicked item position
     */
    fun onItemClick(position: Int)

    /**
     * The long click occurs on entire row
     * @param position the clicked item position
     */
    fun onItemLongClick(position: Int)
}
